Library/Formula/infer.rb

```diff
@@ -0,0 +1,54 @@
+class Infer < Formula
+  desc "Static analyzer for Java, C and Objective-C"
+  homepage "http://fbinfer.com/"
+  url "https://github.com/facebook/infer/releases/download/v0.5.0/infer-osx-v0.5.0.tar.xz"
+  sha256 "6a8547ac0b75a5e2bbeccae2169e39f753a60adbcacb6c94599fd31343a71ce7"
+
+  option "without-clang", "Build without C/Objective-C analyzer"
+  option "without-java", "Build without Java analyzer"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "opam" => :build
+
+  def install
+    if build.without?("clang") && build.without?("java")
+      odie "infer: --without-clang and --without-java are mutually exclusive"
+    end
+    opamroot = buildpath/"build"
```

Comment:
Only cosmetic, but please add an empty line between the check and the beginning of the `opam` stuff.